### ISSUE TITLE
Tagbar will not open on kitty terminal

### ISSUE THREAD

**Issue by neosapien3247** [andreyorst/tagbar.kak#12]
I have had this problem consistently on kitty and I have succeeded in finding the source of the problem.

Just for info:
```
$ kitty --version
kitty 0.21.2 created by Kovid Goyal
```

The problem was very simple: executing the ":tagbar-enable" command did not generate a new window. I tried it later on in xterm and it worked flawlessly, so i checked if it was because kitty wasn't configured properly, and I suspected I needed to have "listen_on" set to something and "allow_remote_control" set to "yes"

(fyi this is my current configuration)
```
font_size				15
font_family				Fira Code
bold_font				auto
italic_font				auto
bold_italic_font		auto
disable_ligatures		never


cursor_shape			beam

open_url_with			brave-browser
detect_urls				yes
enable_audio_bell		no

background_opacity		0.8
foreground				#bcc3cf
background				#17191c
#282c34

editor 					kak
window_padding_width	15
allow_remote_control	yes
listen_on               unix:/tmp/kitty_socket
```

But this didn't seem to change anything (though I still suspect they are necessary for the plugin to use the kitty backend.

So i wnet ahead and had a look at the bash code for the plugin and I debugged until I found the source of the problem:

In the command definition for "tagbar-display" I found that everything worked fine within the if statement for the kitty backend:
```
    elif [ "$TERM" = "xterm-kitty" ]; then
        match=""
        if [ -n "$kak_client_env_KITTY_WINDOW_ID" ]; then
            match="--match=id:$kak_client_env_KITTY_WINDOW_ID"
        fi

        listen=""
        if [ -n "$kak_client_env_KITTY_LISTEN_ON" ]; then
            listen="--to=$kak_client_env_KITTY_LISTEN_ON"
        fi
        kitty @ $listen launch --no-response --type="$kak_opt_kitty_window_type" --cwd="$PWD" $match kak -c ${kak_session} -e "${tagbar_cmd}"
```
Except for the last line which actually launched the kitty terminal, I therefore checked wether I could manually execute this on kitty (by simply echoing the contents of the kitty subcommand to a file and then executinf kitty with said subcommand)
```
...
        if [ -n "$kak_client_env_KITTY_LISTEN_ON" ]; then
            listen="--to=$kak_client_env_KITTY_LISTEN_ON"
        fi
+++ echo @ $listen launch --no-response --type="$kak_opt_kitty_window_type" --cwd="$PWD" $match kak -c ${kak_session} -e "${tagbar_cmd}" >> $HOME/file
        kitty @ $listen launch --no-response --type="$kak_opt_kitty_window_type" --cwd="$PWD" $match kak -c ${kak_session} -e "${tagbar_cmd}"
...
```
I then executed the above code simply by reopening kakoune and running ":tagbar-enable" on a random c project. Then I went to look at the file contents:
```
$ cat ~/file
@ --to=unix:/tmp/kitty_socket-497246 launch --no-response --type=kitty --cwd=/home/neosapien --match=id:1 kak -c 505670 -e try %{ buffer *tagbar* } catch %{ edit! -debug -scratch *tagbar* }
                rename-client %opt{tagbarclient}
                hook -group tagbar-watchers global FocusIn (?!tagbarclient).* %{ try %{ tagbar-update 'focus' } }
                hook -group tagbar-watchers global WinDisplay (?!\*tagbar\*).* %{ try %{ tagbar-update } }
                hook -group tagbar-watchers global BufWritePost (?!\*tagbar\*).* %{ try %{ tagbar-update } }
                hook -group tagbar-watchers global BufOpenFile (?!\*tagbar\*).* %{ try %{ tagbar-update } }
                hook -group tagbar-watchers global BufNewFile (?!\*tagbar\*).* %{ try %{ tagbar-update } }
                hook -group tagbar-watchers global WinSetOption tagbar_(sort|display_anon)=.* %{ try %{ tagbar-update } }
                focus client0
```

I then executed everything except for the kakoune specific commands:
```
$ kitty @ --to=unix:/tmp/kitty_socket-497246 launch --no-response --type=kitty --cwd=/home/neosapien --match=id:1 kak -c 505670
```
and got the following error message:
```
kitty is not a valid value for the --type option. Valid values are: tab, window, os-window, overlay, primary, background, clipboard
```

It turns out that the "$kak_opt_kitty_window_type" variable was outputting "kitty" which wasn't a valid option. I therefore solved the problem by selecting the option that suited my needs best by replacing the variable with "os-window" and it worked like a charm. I also tried the other options for good measure and they all worked as well.

TLDR; Tagbar wasn't opening in kitty and it was solvable by changing one line in the plugin from:
```
kitty @ $listen launch --no-response --type="$kak_opt_kitty_window_type" --cwd="$PWD" $match kak -c ${kak_session} -e "${tagbar_cmd}"
```
to:
```
kitty @ $listen launch --no-response --type="os-window" --cwd="$PWD" $match kak -c ${kak_session} -e "${tagbar_cmd}"
```

**Comment by andreyorst** [comment]
>It turns out that the "$kak_opt_kitty_window_type" variable was outputting "kitty" which wasn't a valid option. I therefore solved the problem by selecting the option that suited my needs best by replacing the variable with "os-window" and it worked like a charm. I also tried the other options for good measure and they all worked as well.

This option is by default set to `window` by Kakoune. It sems that your configuration somehow set it to a wrong value. So instead of changing the plugin code you should change the variable value.

**Comment by neosapien3247** [comment]
Alright, can you tell me where in kakoune  that value is stored?

**Comment by neosapien3247** [comment]
Nevermind I found it:

For reference to anyone who needs to change the value, it's set in the file:
```
/usr/share/kak/rc/windowing/kitty.kak
```

And you just change the option:
```
declare-option -docstring %{window type that kitty creates on new and repl calls (kitty|os)} str kitty_window_type kitty
```

to:
```
declare-option -docstring %{window type that kitty creates on new and repl calls (kitty|os)} str kitty_window_type os-window
```

I don't know why my configuration was this way but hopefully this will help someone else with the problem

**Comment by andreyorst** [comment]
>Nevermind I found it:
>
>For reference to anyone who needs to change the value, it's set in the file:
>```
>/usr/share/kak/rc/windowing/kitty.kak
>```
>
>And you just change the option:
>```
>declare-option -docstring %{window type that kitty creates on new and repl calls (kitty|os)} str kitty_window_type kitty
>```
>
>to:
>```
>declare-option -docstring %{window type that kitty creates on new and repl calls (kitty|os)} str kitty_window_type os-window
>```

There's no need to change source files. You can change options with the `set-option` command from the kakrc file. So it would just be `set-option global kitty_window_type os-window`

>I don't know why my configuration was this way but hopefully this will help someone else with the problem

It was changed in this commit: 

https://github.com/mawww/kakoune/commit/8d5d0d7d909c92ab1fbe6d07152647e4ecd0a564

So you just need to update your Kakoune
